feature_engine/prediction/base_predictor.py

```diff
@@ -0,0 +1,235 @@
+        Whether the bins should of equal width ('equal_width') or equal frequency
+        ('equal_frequency').
+
+    Attributes
+    ----------
+    variables_categorical_:
+        The group of categorical input variables that will be used for prediction.
+
+    variables_numerical_:
+        The group of numerical input variables that will be used for prediction.
+
+    pipeline_:
+        A Sickit-learn Pipeline with a dicretiser and/or encoder. Used to determine the
+        mean target value per category or bin, per variable.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Learn the mean target value per category or per bin, for each variable.
+    predict:
+        Predict using the average of the target mean value across variables.
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+    Adapted from:
+
+    .. [1] Miller, et al. "Predicting customer behaviour: The University of Melbourne’s
+        KDD Cup report". JMLR Workshop and Conference Proceeding. KDD 2009
+        http://proceedings.mlr.press/v7/miller09/miller09.pdf
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        bins: int = 5,
+        strategy: str = "equal_width",
+    ):
+        # boolean value can be interpreted as an integer
+        if not isinstance(bins, int) or isinstance(bins, bool):
+            raise ValueError(f"bins must be an integer. Got {bins} instead.")
+
+        if strategy not in ["equal_width", "equal_frequency"]:
+            raise ValueError(
+                "strategy takes only values equal_width or equal_frequency. Got "
+                f"{strategy} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.bins = bins
+        self.strategy = strategy
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Learn the mean target value per category or bin.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : pandas series of shape = [n_samples,]
+            The target variable.
+        """
+        # check if 'X' is a dataframe
+        _is_dataframe(X)
+
+        # identify categorical and numerical variables
+        (
+            self.variables_categorical_,
+            self.variables_numerical_,
+        ) = _find_categorical_and_numerical_variables(X, self.variables)
+
+        # check for missing values
+        _check_contains_na(X, self.variables_numerical_)
+        _check_contains_na(X, self.variables_categorical_)
+
+        # check inf
+        _check_contains_inf(X, self.variables_numerical_)
+
```

Comment:
if we use Check_X-y we do not need to check for na and inf ourselves. In theory. But I have not tried.

I was wondering if you could try commenting this out and see if the tests still pass?

Reply:
See prior post.